src/sbml/conversion/SBMLRateRuleConverter.cpp

```diff
@@ -813,7 +893,133 @@ SBMLRateRuleConverter::reconstructModel()
+
+    // if right child is not a variable species, it's not -x+y
+    ASTNode* rightChild = node->getRightChild();
+    if (!isVariableSpecies(rightChild, model))
+        return false;
+
+    // if we get to this point, the only thing left to check is 
+    // whether the ->left->right grandchild (the x in -x+y) is a variable species.
+    ASTNode* grandChild = leftChild->getRightChild();
+    return isVariableSpecies(grandChild, model);
+}
+
+// helper function to determine whether a node is the second "-" in a k-x-y expression, where x,y are variable species and k is a numerical constant or a parameter
+bool SBMLRateRuleConverter::isKMinusXMinusY(ASTNode* node, Model* model)
+{
+    // if node is not binary, it's not k-x-y
+    if (node->getNumChildren() != 2)
+        return false;
+
+    // if node is not a minus, it's not the second "-" in k-x-y
+    if (node->getType() != ASTNodeType_t::AST_MINUS)
+        return false;
+
+    // if right child is not a variable species, it's not k-x-y
+    ASTNode* rightChild = node->getRightChild();
+    if(!isVariableSpecies(rightChild, model))
+        return false;
+
+    // if left child is not a minus, it's not k-x-y
+    ASTNode* leftChild = node->getLeftChild();
+    if (leftChild->getType() != ASTNodeType_t::AST_MINUS)
+        return false;
+
+    // if left child is not binary, it's not k-x-y
+    if (leftChild->getNumChildren() != 2)
+        return false;
+
+    // if we've gotten this far, what's left to check are the children of the left child (k and x)
+    ASTNode* k = leftChild->getLeftChild();
+    ASTNode* x = leftChild->getRightChild();
+    return isNumericalConstantOrParameter(k, model) && isVariableSpecies(x, model);
+}
+
+// check whether a node is the minus sign in a k-x expression
+bool SBMLRateRuleConverter::isKMinusX(ASTNode* node, Model* model)
+{
+    // if node is not binary, it's not k-x
+    if (node->getNumChildren() != 2)
+        return false;
+
+    // if node is not a minus, it's not "-" in k-x
+    if (node->getType() != ASTNodeType_t::AST_MINUS)
+        return false;
+
+    // if right child is not a variable species, it's not k-x
+    ASTNode* rightChild = node->getRightChild();
+    if (!isVariableSpecies(rightChild, model))
+        return false;
+
+    // if left child is not a numerical constant or a parameter, it's not k-x - else it is!
+    ASTNode* leftChild = node->getLeftChild();
+    return isNumericalConstantOrParameter(leftChild, model);
+}
 
+bool SBMLRateRuleConverter::isVariableSpecies(ASTNode* node, Model* model)
+{
+    if (!node->isName()) // some nodes, like * operators, don't seem to have a name in the first place
+        return false;
+    Species* species = model->getSpecies(node->getName());
+    Parameter* parameter = model->getParameter(node->getName()); // some species in rate rules may be defined as variable parameters
+    bool isSpecies = (species != NULL || parameter != NULL);
+    bool isVariable = !node->isConstant();
```

Comment:
the node being constant is not a clear indicator
you will need to check whether the species or parameter isConstant = false